sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -766,6 +768,23 @@ object CollapseProject extends Rule[LogicalPlan] {
     })
   }
```

Comment:
We could extend to other cases like `case p @ Project(_, agg: Aggregate)`, but leave it untouched for now.